src/update_helper.rs

```diff
@@ -0,0 +1,199 @@
+//! # Functions to update timestamps.
+
+use crate::chat::{Chat, ChatId};
+use crate::contact::Contact;
+use crate::context::Context;
+use crate::param::{Param, Params};
+use anyhow::Result;
+
+impl Context {
+    /// Updates a contact's timestamp, if reasonable.
+    /// Returns true if the caller shall update the settings belonging to the scope.
+    /// (if we have a ContactId type at some point, the function should go there)
+    pub(crate) async fn update_contacts_timestamp(
+        &self,
+        contact_id: u32,
+        scope: Param,
+        new_timestamp: i64,
+    ) -> Result<bool> {
+        if let Ok(mut contact) = Contact::load_from_db(self, contact_id).await {
+            if contact.param.set_timestamp(scope, new_timestamp)? {
+                contact.update_param(self).await?;
+                return Ok(true);
+            }
+        }
+        Ok(false)
+    }
+}
+
+impl ChatId {
+    /// Updates a chat id's timestamp on disk, if reasonable.
+    /// Returns true if the caller shall update the settings belonging to the scope.
+    pub(crate) async fn update_timestamp(
+        &self,
+        context: &Context,
+        scope: Param,
+        new_timestamp: i64,
+    ) -> Result<bool> {
+        if let Ok(mut chat) = Chat::load_from_db(context, *self).await {
```

Comment:
Same as for contact, failure to load the chat from the database should be bubbled up.